strawberryfields/utils/post_processing.py

```diff
@@ -0,0 +1,271 @@
+# Copyright 2019-2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This module provides functions for post-processing samples.
+"""
+import numpy as np
+
+
+def _product_for_modes(samples, modes=None):
+    """Getting the product of samples across modes.
+Args:
+        samples (ndarray): the photon number samples with a shape
+            of (shots, modes)
+        modes (Sequence): a flat sequence containing indices of modes to get
+            the expectation value for
+
+    Returns:
+        ndarray: product of the samples across modes
+    """
+    modes = np.array(modes)
+    selected_samples = samples[:, modes]
+    return np.prod(selected_samples, axis=1)
+
+
+def _check_samples(samples):
+    """Validation check for the input samples.
+
+    Checks include data types checks and dimension of the input samples.
+
+    Args:
+        samples (ndarray): the photon number samples with a shape of (shots,
+            modes)
```

Comment:
For consistency, we tend to use ``array`` rather than ``ndarray`` in the rest of the code base.
```suggestion
        samples (array): the photon number samples with a shape of ``(shots, modes)``
```